assets/src/edit-story/components/inspector/prepublish/checklistTab.js

```diff
@@ -194,14 +219,70 @@ const ChecklistTab = (props) => {
     [checklist]
   );
 
+  const selectElement = useCallback(
+    ({ elementId, elements, pageId }) => {
+      if (pageId) {
+        setCurrentPage({ pageId });
+      }
+      if (Array.isArray(elements)) {
+        setSelectedElementsById({
+          elementIds: elements.map(({ id }) => id),
+        });
+      } else if (elementId) {
+        setSelectedElementsById({ elementIds: [elementId] });
+      }
+    },
+    [setCurrentPage, setSelectedElementsById]
+  );
+
+  const getOnClick = useCallback(
+    ({ elementId, pageId, elements }) => {
+      if (!elementId && !pageId && !elements) {
+        return undefined;
+      }
+      return () => selectElement({ elementId, pageId, elements });
+    },
+    [selectElement]
+  );
+
+  const getHandleKeyPress = useCallback(
+    ({ elementId, elements, pageId }) => {
+      if (!elementId && !pageId && !elements) {
+        return undefined;
+      }
+      return (event) => {
+        if (event.key === 'Enter') {
+          selectElement({ elementId, elements, pageId });
+        }
+      };
+    },
+    [selectElement]
+  );
+
   const renderRow = useCallback(
-    ({ message, help, id, pageGroup }) => (
-      <Row key={`guidance-${id}`} pageGroup={pageGroup}>
-        {message}
-        <HelperText>{help}</HelperText>
-      </Row>
-    ),
-    []
+    ({ message, help, id, pageGroup, ...args }) => {
+      const onClick = getOnClick(args);
+      const handleKeyPress = getHandleKeyPress(args);
+      return (
+        <Row
+          tabIndex={0}
+          onClick={onClick}
```

Comment:
Sorry for all the back and forth on this PR! But we should get this right.

I'm not a fan of this using `onClick` but not the correct role. VoiceOver announces it as clickable, but I'm not sure other screen readers will do that as well.

[ARIA docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role) say "the `button` role should be used for clickable elements that trigger a response when activated by the user."

I think if you make it a proper button this should also fix the `onKeyDown` issue.